File: Pages/LoginPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using System;

public class LoginPage
{
    private readonly IWebDriver driver;
    private readonly WebDriverWait wait;

    public LoginPage(IWebDriver driver)
    {
        this.driver = driver;
        wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));
    }

    private IWebElement EmailField => wait.Until(driver => driver.FindElement(By.XPath("//input[@placeholder='Email address']")));
    private IWebElement PasswordField => driver.FindElement(By.XPath("//input[@placeholder='Password']"));
    private IWebElement LoginButton => driver.FindElement(By.XPath("//input[@id='login-id']"));

    public void EnterEmail(string email)
    {
        EmailField.SendKeys(email);
    }

    public void EnterPassword(string password)
    {
        PasswordField.SendKeys(password);
    }

    public void ClickLogin()
    {
        LoginButton.Click();
    }

    public void Login(string email, string password)
    {
        EnterEmail(email);
        EnterPassword(password);
        ClickLogin();
    }
}

File: pages/login.page.ts
import { type Page, type Locator } from '@playwright/test';

/**
 * Page Object Model for the Login page.
 * Handles authentication via email and password form.
 */
export class LoginPage {
  // Locators
  readonly emailField: Locator;
  readonly passwordField: Locator;
  readonly loginButton: Locator;

  constructor(private page: Page) {
    this.emailField = page.locator("input[placeholder='Email address']");
    this.passwordField = page.locator("input[placeholder='Password']");
    this.loginButton = page.locator('input#login-id');
  }

  /**
   * Enter email address into the email field.
   */
  async enterEmail(email: string): Promise<void> {
    await this.emailField.fill(email);
  }

  /**
   * Enter password into the password field.
   */
  async enterPassword(password: string): Promise<void> {
    await this.passwordField.fill(password);
  }

  /**
   * Click the login button.
   */
  async clickLogin(): Promise<void> {
    await this.loginButton.click();
  }

  /**
   * Perform a complete login with the given credentials.
   */
  async login(email: string, password: string): Promise<void> {
    await this.enterEmail(email);
    await this.enterPassword(password);
    await this.clickLogin();
  }
}
